src/rasqal/src/runtime.rs

```diff
@@ -608,13 +608,13 @@ impl QuantumRuntime {
 
           // TODO: Make smart-pointers forward operators.
           let result = Ptr::from(match op {
-            Operator::Multiply => left.deref() * right.deref(),
-            Operator::Divide => left.deref() / right.deref(),
-            Operator::Add => left.deref() + right.deref(),
-            Operator::Subtract => left.deref() - right.deref(),
-            Operator::Or => left.deref() | right.deref(),
-            Operator::And => left.deref() & right.deref(),
-            Operator::Xor => left.deref() ^ right.deref(),
+            Operator::Multiply => left * right,
+            Operator::Divide => left / right,
+            Operator::Add => left + right,
+            Operator::Subtract => left - right,
+            Operator::Or => left | right,
+            Operator::And => left & right,
+            Operator::Xor => left ^ right,
             Operator::PowerOf => Value::from(left.as_int().pow(right.as_int() as u32))
```

Comment:
why can't you implement PowerOf?

Reply:
Pretty sure it's not a built-in operator but I'll double-check.